app/src/main/java/protect/card_locker/ScanActivity.kt

```diff
@@ -1,541 +1,548 @@
-            builder.setTitle(getString(R.string.add_a_card_in_a_different_way));
-            builder.setAdapter(
-                    adapter,
-                    (dialogInterface, i) -> {
-                        switch (i) {
-                            case 0:
-                                addWithoutBarcode();
-                                break;
-                            case 1:
-                                addManually();
-                                break;
-                            case 2:
-                                addFromImage();
-                                break;
-                            case 3:
-                                addFromPdf();
-                                break;
-                            case 4:
-                                addFromPkPass();
-                                break;
-                            default:
-                                throw new IllegalArgumentException("Unknown 'Add a card in a different way' dialog option");
-                        }
-                    }
-            );
-            builder.setOnCancelListener(dialogInterface -> setScannerActive(true));
-            builder.show();
-        });
+            val adapter: ListAdapter = SimpleAdapter(
+                    this,
+                    list,
+                    R.layout.alertdialog_row_with_icon,
+                    columns,
+                    intArrayOf(R.id.textView, R.id.imageView)
+            )
+
+            val builder = MaterialAlertDialogBuilder(this).apply {
+                setTitle(getString(R.string.add_a_card_in_a_different_way))
+                setAdapter(adapter) { _, i ->
+                        when (i) {
+                    0 -> addWithoutBarcode()
+                    1 -> addManually()
+                    2 -> addFromImage()
+                    3 -> addFromPdf()
+                    4 -> addFromPkPass()
+                        else -> throw IllegalArgumentException(
+                            "Unknown 'Add a card in a different way' dialog option: $i"
+                    )
+                }
+                }
+                setOnCancelListener { _ -> setScannerActive(true) }
+            }
+            builder.show()
+        }
 
         // Configure barcodeScanner
-        barcodeScannerView = binding.zxingBarcodeScanner;
-        Intent barcodeScannerIntent = new Intent();
-        Bundle barcodeScannerIntentBundle = new Bundle();
-        barcodeScannerIntentBundle.putBoolean(DecodeHintType.ALSO_INVERTED.name(), Boolean.TRUE);
-        barcodeScannerIntent.putExtras(barcodeScannerIntentBundle);
-        barcodeScannerView.initializeFromIntent(barcodeScannerIntent);
+        barcodeScannerView = binding.zxingBarcodeScanner
+        val barcodeScannerIntent = Intent().apply {
+            putExtra(DecodeHintType.ALSO_INVERTED.name, java.lang.Boolean.TRUE)
+        }
+        barcodeScannerView.initializeFromIntent(barcodeScannerIntent)
```

Comment:
Doesn't this change the meaning of this code? I believe the DecodeHintType should be part of the bundle inside the intent, not in the intent itself. But I haven't tested if this breaks scanning inverted codes like
<img width="256" height="256" alt="image" src="https://github.com/user-attachments/assets/54479ec2-9718-45dd-a5b5-77b7d959bee2" />

(please ignore the link in the QR code, it's just one of the fastest examples I could find)